ql/src/java/org/apache/hadoop/hive/ql/ddl/table/info/show/status/formatter/ShowTableStatusFormatter.java

```diff
@@ -89,7 +93,18 @@ List<Path> getLocations(Hive db, Partition partition, Table table) throws HiveEx
     List<Path> locations = new ArrayList<Path>();
     if (table.isPartitioned()) {
       if (partition == null) {
-        for (Partition currPartition : db.getPartitions(table)) {
+      List<Partition> partitions;
+      GetProjectionsSpec getProjectionsSpec = new GetPartitionProjectionsSpecBuilder()
+              .addProjectFieldList(Arrays.asList("sd.location")).build();
+        GetPartitionsRequest request = new GetPartitionsRequest(table.getDbName(), table.getTableName(), getProjectionsSpec, null);
+        request.setCatName(table.getCatName());
+        try{
+          partitions = db.getPartitionsWithSpecs(table, request);
```

Comment:
why not use new helper method from `PartitionUtils`?

Reply:
PartitionUtils is a part of org.apache.hadoop.hive.ql.ddl.table.partition
package implying it is made for DDL operations for partitioned tables. Here we are not performing any DDL operation.

Reply:
`PartitionUtils#getPartitionsWithSpecs` is a public method, so I don't get what is an issue. 
Anyways that's fine